modules/auxiliary/admin/mssql/mssql_idf.rb

```diff
@@ -104,11 +104,16 @@ def run
     widths = [0, 0, 0, 0, 0, 9]
     total_width = 0
 
-    (column_data|headings).each { |row|
-      0.upto(4) { |col|
-        widths[col] = row[col].length if row[col].length > widths[col]
+    begin
+      (column_data|headings).each { |row|
+        0.upto(4) { |col|
+          widths[col] = row[col].length if row[col].length > widths[col]
+        }
       }
-    }
+    rescue NoMethodError
+      print_error("No columns matched the pattern #{datastore['NAMES'].inspect}. Update the NAMES option to change this search pattern.")
+      return
+    end
 
     widths.each { |a|
```

Comment:
This is scope creep, but a separate PR to update this to use rex-table might be nice

Reply:
Sure, I'll look into that.